### PR TITLE
Add tests for the assembler object (and avoid misusing string{} for u256)

### PR DESCRIPTION
**Pull request by axic**
Part of #1275.

Closes #4682.

### CODE REVIEW

File: libevmasm/Assembly.cpp [hunk 1]
```diff
@@ -264,7 +264,7 @@ Json::Value Assembly::assemblyJSON(StringMap const& _sourceCodes) const
 					createJsonValue("PUSH [ErrorTag]", i.location().start, i.location().end, ""));
 			else
 				collection.append(
-					createJsonValue("PUSH [tag]", i.location().start, i.location().end, string(i.data())));
```
Comment (chriseth) [inline libevmasm/Assembly.cpp:267]: Oh shoot - this was modified recently?

Comment (chriseth) [reply libevmasm/Assembly.cpp:267]: Ah sorry, confused it with the one below. What is the difference between the two versions?

**Review by chriseth — APPROVED** [review]

File: libevmasm/Assembly.cpp (hunk 1 above)
Comment (axic) [reply libevmasm/Assembly.cpp:267]: There's one we use (PushTag) and there's the (PushString) which is only used by LLL right now.

`case Tag` is for the jumpdest, `case Pushtag` for the jump` and `case PushString` for a feature to store literals in code.

**Comment by axic** [comment]
The mac build experienced a hiccup today (restarted runs on several PRs multiple times today): `Error: Failed to download resource "boost"`

**Comment by axic** [comment]
Merging this now, because the Mac failure is a CI problem.

**Pull request merged**